Pod/Classes/LSClockState.m

```diff
@@ -149,45 +155,45 @@ - (void)update {
 
 #pragma mark - Private
 
-- (void)_persistToUserDefaults {
+- (void)_persistSamplesToUserDefaults:(NSArray *)samples {
     NSData *data = [NSKeyedArchiver archivedDataWithRootObject:@{
         kTimestampMicrosKey: @([LSClockState nowMicros]),
-        kSamplesKey: self.samples
+        kSamplesKey: samples
     }];
     [[NSUserDefaults standardUserDefaults] setObject:data forKey:kUserDefaultsKey];
 }
 
 // Overwrites all state; only intended to be called from init.
 - (void)_tryToRestoreFromUserDefaults {
-    self.samples = [NSMutableArray array];
-    self.currentOffsetMicros = 0;
-    self.currentOffsetAge = kMaxOffsetAge + 1;
-    NSData *data = [[NSUserDefaults standardUserDefaults] objectForKey:kUserDefaultsKey];
-    if (data != nil) {
-        // Check out this gem, which ends with an emphatic "*Do not use NSKeyedArchiver*":
-        // http://stackoverflow.com/a/17301208/3399080
-        @try {
+    dispatch_sync(self.samplesQueue, ^{
+        self.currentOffsetMicros = 0;
+        self.currentOffsetAge = kMaxOffsetAge + 1;
+        NSData *data = [[NSUserDefaults standardUserDefaults] objectForKey:kUserDefaultsKey];
+        if (data != nil) {
+            // Check out this gem, which ends with an emphatic "*Do not use NSKeyedArchiver*":
+            // http://stackoverflow.com/a/17301208/3399080
             NSDictionary *dict = [NSKeyedUnarchiver unarchiveObjectWithData:data];
             NSNumber *tsMicros = [dict objectForKey:kTimestampMicrosKey];
             NSArray *samples = [dict objectForKey:kSamplesKey];
             SInt64 nowMicros = [LSClockState nowMicros];
             if (dict && tsMicros && samples && (tsMicros.longLongValue > (nowMicros - kStoredSamplesTTLMicros)) &&
                 (tsMicros.longLongValue < nowMicros) /* <-- sanity check */) {
-                NSUInteger loc = MAX(0, samples.count - (kMaxOffsetAge + 1));
+                NSUInteger loc = MAX(0, (NSInteger)samples.count - (kMaxOffsetAge + 1));
                 NSUInteger len = samples.count - loc;
-                self.samples = [NSMutableArray arrayWithArray:[samples subarrayWithRange:NSMakeRange(loc, len)]];
+                self.samples = [samples subarrayWithRange:NSMakeRange(loc, len)].mutableCopy;
             }
-        } @catch (NSException *e) {
-            NSLog(@"Unable to decode LSClockState data. Leaving things be.");
+        } else {
+            self.samples = [NSMutableArray new];
```

Comment:
For reviewer: I've replaces a few usages of [NSArray array...] with `new`, `copy` or `mutableCopy`.
The former one creates an object instance and places it on an autorelease pool. The latter ones are creating a reference and releasing it immediately after assigning to the property. Autoreleased references should be avoided whenever you have control.